<h1 align="center">Title: Indian Startup Funding Analysis</h1>

## Business Objective:
The goal of this project is to investigate the Indian Startup Ecosystem to better understand and provide valuable insight into the opportunites and challenges in the ecosystem in order to help stakeholders who plan on venturing into the startup ecosystem in India to make informed decisions based on findings from analyzing the dataset from 2018 to 2021.

### Import all necessary libraries

In [2]:
# Data Manipulation Libraries
import numpy as np
import pandas as pd
from sklearn.impute import SimpleImputer
import re

# Visualization Libraries
import matplotlib.pyplot as plt
import seaborn as sns

# Database Libraries
import pyodbc

# Other Utilities
from dotenv import dotenv_values
from warnings import filterwarnings
filterwarnings('ignore')

### Create connection for SQL Server

In [3]:
# Load Environment Variables
config = dotenv_values('.env')

Server_name = config.get('Server')
Database_name = config.get('Database')
Username = config.get('Login')
PassWord = config.get('Password')

# Create Database Connection
connection_string = f"DRIVER={{SQL Server}};SERVER={Server_name};DATABASE={Database_name};UID={Username};PWD={PassWord}"

connection = pyodbc.connect(connection_string)

### Load tables from SQL and save a copy of the dataset

In [4]:
# Query table as dataframe
query = "SELECT * FROM dbo.LP1_startup_funding2020"
df_2020 = pd.read_sql(query, connection)

# Preview the dataset
df_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  
0                      Angel investors  200000.0      None     None  
1                      GSF Accelerator  100000.0  Pre-seed     None  
2                       Venture Center       NaN  Pre-seed     None  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  
4  Siana Capital Management, Info Edge  340000.0      None     None

In [5]:
df_2020.shape

(1055, 10)

In [6]:
# Query table as dataframe
query = "SELECT * FROM dbo.LP1_startup_funding2021"
df_2021 = pd.read_sql(query, connection)

# Preview the dataset
df_2021.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [7]:
df_2021.shape

(1209, 9)

In [8]:
# Save the dataset as csv files for each year
# df_2021.to_csv('startup_funding2021.csv')
# df_2020.to_csv('startup_funding2020.csv')

In [9]:
df_2021.shape

(1209, 9)

### Load other dataset in csv file

In [10]:
# Read csv file
df_2018 = pd.read_csv('startup_funding2018.csv')
df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [11]:
df_2018.shape

(526, 6)

In [12]:
# Read csv file
df_2019 = pd.read_csv('startup_funding2019.csv')
df_2019.head()

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [13]:
df_2019.shape

(89, 9)

<h3>Hypothesis</h3>
<p>Null: Tech startups do not receive enough funding from investors.</p>
<p>Alternative: Tech startups receive enough funding from investors.</p>

<h3>Questions</h3>

1. What is the total amount of funding received from investors from 2018 to 2021?
2. How many startups emerged from 2018 to 2021?
3. What is the level of funding for startups based on their sector?
4. How dispersed are the startup firms over the locations?
5. What is the trend of funding based on the years of funding?
6. Is there a relationship between stage and funding receive?

## Rename columns, generate columns where needed and merge datasets

In [14]:
# Add Funding Year for all the dataset
df_2021['Funding_Year']=2021; df_2020['Funding_Year']=2020; df_2019['Funding_Year']=2019; df_2018['Funding_Year']=2018

In [15]:
# Adjust the column names of the columns in the datasets

df_2021.rename(columns={'Amount': 'Amount($)'},inplace=True)

df_2020.rename(columns={'Amount': 'Amount($)'},inplace=True)

df_2019.rename(columns={'Company/Brand': 'Company_Brand',
                        'What it does': 'What_it_does'}, inplace=True)

df_2018.rename(columns={'Company Name': 'Company_Brand',
                'Amount': 'Amount($)',
                'About Company': 'What_it_does',
                'Round/Series': 'Stage'}, inplace=True)

In [16]:
df_2021.head(1)

Company_Brand  Founded HeadQuarter      Sector  \
0  Unbox Robotics   2019.0   Bangalore  AI startup   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   

                      Founders                     Investor   Amount($)  \
0  Pramod Ghadge, Shahid Memon  BEENEXT, Entrepreneur First  $1,200,000   

          Stage  Funding_Year  
0  Pre-series A          2021

In [17]:
df_2020.head(1)

Company_Brand  Founded HeadQuarter    Sector                  What_it_does  \
0    Aqgromalin   2019.0     Chennai  AgriTech  Cultivating Ideas for Profit   

                          Founders         Investor  Amount($) Stage column10  \
0  Prasanna Manogaran, Bharani C L  Angel investors   200000.0  None     None   

   Funding_Year  
0          2020

In [18]:
df_2019.head(1)

Company_Brand  Founded HeadQuarter     Sector  \
0  Bombay Shaving      NaN         NaN  Ecommerce   

                                 What_it_does            Founders  \
0  Provides a range of male grooming products  Shantanu Deshpande   

               Investor   Amount($) Stage  Funding_Year  
0  Sixth Sense Ventures  $6,300,000   NaN          2019

In [19]:
df_2018.head(1)

Company_Brand                                           Industry Stage  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...  Seed   

  Amount($)                     Location  \
0    250000  Bangalore, Karnataka, India   

                                        What_it_does  Funding_Year  
0  TheCollegeFever is a hub for fun, fiesta and f...          2018

Observing a preview of all the dataset, we will consider the 2021 dataset as our base format the columns in all our dataset. The 2018 dataset do not follow the agreed base dataset and must be corrected to meet the standard. We see for instance from the Location column that the value in the first index position matches the values for HeadQuarters in the other datasets. The Location column in the 2018 dataset must be dropped and the industry column name changed to Sector.

### Extract HeadQuarter from Location column in the 2018 dataset

In [20]:
# Add HeadQuarter column from the location column
df_2018['HeadQuarter'] = df_2018['Location'].str.split(',').str[0]
df_2018.head()

Company_Brand                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage    Amount($)                          Location  \
0      Seed       250000       Bangalore, Karnataka, India   
1      Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2  Series A  ₹65,000,000           Gurgaon, Haryana, India   
3     Angel      2000000       Noida, Uttar Pradesh, India   
4      Seed            —  Hyderabad, Andhra Pradesh, India   

                                        What_it_does  Funding_Year HeadQuarter  
0  TheCollegeFever is a hub for fun, fiesta and f...          2018   Bangalore  
1  A startup which aggregates milk from dairy far...          2018      Mumbai  
2          Leading Online Loans Marketplace in India          2018     Gurgaon  
3  PayMe India is an innovative FinTech organizat...          2018       Noida  
4  Eunimart is a one stop solution for merchants ...          2018   Hyderabad

In [21]:
df_2018.rename(columns={'Industry':'Sector'}, inplace=True)

df_2018.drop('Location', axis=1, inplace=True)

df_2018.head()

Company_Brand                                             Sector  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

      Stage    Amount($)                                       What_it_does  \
0      Seed       250000  TheCollegeFever is a hub for fun, fiesta and f...   
1      Seed  ₹40,000,000  A startup which aggregates milk from dairy far...   
2  Series A  ₹65,000,000          Leading Online Loans Marketplace in India   
3     Angel      2000000  PayMe India is an innovative FinTech organizat...   
4      Seed            —  Eunimart is a one stop solution for merchants ...   

   Funding_Year HeadQuarter  
0          2018   Bangalore  
1          2018      Mumbai  
2          2018     Gurgaon  
3          2018       Noida  
4          2018   Hyderabad

Concatenate the dataset into one full dataset

In [22]:
full_df = pd.concat([df_2021, df_2020, df_2019, df_2018],ignore_index=True)
full_df.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor     Amount($)         Stage  \
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A   
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None   
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D   
3                         CDC Group, IDG Capital   $51,000,000      Series C   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed   

   Funding_Year column10  
0          2021      NaN  
1          2021      NaN  
2          2021      NaN  
3          2021      NaN  
4          2021      NaN

## Exploratory Data Analysis (EDA)

In [23]:
print('The rows and columns in the dataset are', full_df.shape, 'respectively.')

The rows and columns in the dataset are (2879, 11) respectively.


In [24]:
# Check structure of the dataset
full_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2879 entries, 0 to 2878
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  2879 non-null   object 
 1   Founded        2110 non-null   float64
 2   HeadQuarter    2765 non-null   object 
 3   Sector         2861 non-null   object 
 4   What_it_does   2879 non-null   object 
 5   Founders       2334 non-null   object 
 6   Investor       2253 non-null   object 
 7   Amount($)      2622 non-null   object 
 8   Stage          1941 non-null   object 
 9   Funding_Year   2879 non-null   int64  
 10  column10       2 non-null      object 
dtypes: float64(1), int64(1), object(9)
memory usage: 247.5+ KB


We see the datatype for each column in the dataset and the number of non-null values in each column.

<h3>Check for missing values</h3>

In [25]:
# Check for total missing values in the columns
full_df.isna().sum()

Company_Brand       0
Founded           769
HeadQuarter       114
Sector             18
What_it_does        0
Founders          545
Investor          626
Amount($)         257
Stage             938
Funding_Year        0
column10         2877
dtype: int64

* We see the total number of missing values in the columns.
* Handling of missing values will be done in the data cleaning phase when the individual columns are inspected.

<h3>Check for duplicate values</h3>

In [26]:
f'There are {full_df.duplicated().sum()} duplicate values in the 2021 dataset'

'There are 23 duplicate values in the 2021 dataset'

* Handing of duplicate values will be done in the data cleaning phase when the  individual columns are inspected.

<h3>Check the descriptive statistics of all the columns in each dataset</h3>

In [27]:
full_df.describe(include=['number', 'object']).T

count unique                               top freq  \
Company_Brand    2879   2214                          BharatPe   10   
Founded        2110.0    NaN                               NaN  NaN   
HeadQuarter      2765    141                         Bangalore  866   
Sector           2861    873                           FinTech  173   
What_it_does     2879   2691  Provides online learning classes    5   
Founders         2334   1980  Ashneer Grover, Shashvat Nakrani    7   
Investor         2253   1777         Inflection Point Ventures   36   
Amount($)        2622    774                                 —  148   
Stage            1941     75                              Seed  606   
Funding_Year   2879.0    NaN                               NaN  NaN   
column10            2      2                          Pre-Seed    1   

                      mean       std     min     25%     50%     75%     max  
Company_Brand          NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Founded        2016.079621  4.368006  1963.0  2015.0  2017.0  2019.0  2021.0  
HeadQuarter            NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Sector                 NaN       NaN     NaN     NaN     NaN     NaN     NaN  
What_it_does           NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Founders               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Investor               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Amount($)              NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Stage                  NaN       NaN     NaN     NaN     NaN     NaN     NaN  
Funding_Year   2020.023619  1.086974  2018.0  2020.0  2020.0  2021.0  2021.0  
column10               NaN       NaN     NaN     NaN     NaN     NaN     NaN

<h4>From the descriptive tables above, we are able to observe the minimum, maximum, and other statistics of the columns. The categorical columns however do not have most of these values as we cannot get most of the statistical measures on categorical values. Further investigation on the columns will be proceeded with in order to get a wholistic view of the dataset.</h4>

## Data Cleaning

### Correct duplicates in the dataset

In [28]:
f'We have {full_df.duplicated().sum()} duplicated rows in the dataset and since there are no unique identifiers assigned to the values we drop the duplicated values.'

'We have 23 duplicated rows in the dataset and since there are no unique identifiers assigned to the values we drop the duplicated values.'

In [29]:
# Correct duplicates in the entire dataset
full_df.drop_duplicates(keep='first', inplace=True)

full_df.duplicated().sum()

0

### Drop Column 10 from the dataset

Considering the number of missing values in the column, imputing is with any value will be in-valid hence we must drop the column.

In [30]:
# Drop column10
full_df.drop('column10', axis=1, inplace=True)
full_df.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor     Amount($)         Stage  \
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A   
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None   
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D   
3                         CDC Group, IDG Capital   $51,000,000      Series C   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed   

   Funding_Year  
0          2021  
1          2021  
2          2021  
3          2021  
4          2021

### Check the amount column for inconsistencies

In [31]:
# Check the unique values in the column
full_df['Amount($)'].unique()

array(['$1,200,000', '$120,000,000', '$30,000,000', '$51,000,000',
       '$2,000,000', '$188,000,000', '$200,000', 'Undisclosed',
       '$1,000,000', '$3,000,000', '$100,000', '$700,000', '$9,000,000',
       '$40,000,000', '$49,000,000', '$400,000', '$300,000',
       '$25,000,000', '$160,000,000', '$150,000', '$1,800,000',
       '$5,000,000', '$850,000', '$53,000,000', '$500,000', '$1,100,000',
       '$6,000,000', '$800,000', '$10,000,000', '$21,000,000',
       '$7,500,000', '$26,000,000', '$7,400,000', '$1,500,000',
       '$600,000', '$800,000,000', '$17,000,000', '$3,500,000',
       '$15,000,000', '$215,000,000', '$2,500,000', '$350,000,000',
       '$5,500,000', '$83,000,000', '$110,000,000', '$500,000,000',
       '$65,000,000', '$150,000,000,000', '$300,000,000', '$2,200,000',
       '$35,000,000', '$140,000,000', '$4,000,000', '$13,000,000', None,
       '$Undisclosed', '$2000000', '$800000', '$6000000', '$2500000',
       '$9500000', '$13000000', '$5000000', '$8000000',

* We consider the base value for the amount column to be dollars so any value in the amount column with no sign will be treated as a dollar value.
* Certain rows in the amount column have wrong inputs which are values of other columns and these rows must also be corrected.
* The undisclosed values in the column should be replaced with NaN since we can't replace it with zero as that will mean the amount raised was 0 dollars.
* The rows with INR currency sign will be converted using the exchange rate for the year in which funding was received.
* The column must be converted to a float dtype after correcting the inconsistencies

In [32]:
# Check for null values in the Amount column
f"There are {full_df['Amount($)'].isnull().sum()} total missing values in the amount column."

'There are 256 total missing values in the amount column.'

The missing values in the amount column cannot be replace with the mean or with zeros. Dealing with these missing values, the assumption is that all the missing values are undisclosed values in our dataset and hence we can drop these values.

In [33]:
# Drop missing values in the dataset where the Amount is null
full_df.dropna(axis=0, subset='Amount($)', inplace=True)

In [34]:
# Check missing values in the columns of the dataset
full_df.isnull().sum()

Company_Brand      0
Founded          700
HeadQuarter       97
Sector            16
What_it_does       0
Founders         539
Investor         612
Amount($)          0
Stage            787
Funding_Year       0
dtype: int64

In [35]:
# Assign the index of the rows to a string
symbols_string = full_df.index[full_df['Amount($)'].isin(['$','—'])]
symbols = ['$','—']

# Replace the columns with a dollar sign or ' in the amount value with NA
full_df.loc[symbols_string, ['Amount($)']] = full_df.loc[symbols_string, ['Amount($)']].replace(symbols, pd.NA, inplace=True)

In [36]:
# Preview rows with amount vales showing $Undisclosed, $undisclosed, and Undisclosed
undisclosed_list = ['$Undisclosed', '$undisclosed', 'Undisclosed']

# Get the index for all rows with undisclosed
undisclosed_index = full_df.index[full_df['Amount($)'].isin(undisclosed_list)]

# Replace undisclosed values with NA
full_df.loc[undisclosed_index, ['Amount($)']] = full_df.loc[undisclosed_index, ['Amount($)']].replace(undisclosed_list, pd.NA)

In [37]:
# Change the values in the amount column with the original amount values in the Stage column
amount_list = ['ITO Angel Network, LetsVenture', 'JITO Angel Network, LetsVenture', 'ah! Ventures', 'Upsparks']

amount_index = full_df.index[full_df['Amount($)'].isin(amount_list)]

stage_values = full_df.loc[amount_index, ['Amount($)']].values

full_df.loc[amount_index, ['Amount($)']] = full_df.loc[amount_index, ['Stage']].values

full_df.loc[amount_index, ['Stage']] = stage_values
full_df.loc[amount_index]

Company_Brand  Founded     HeadQuarter           Sector  \
98            FanPlay   2020.0  Computer Games   Computer Games   
538       Little Leap   2020.0       New Delhi           EdTech   
551             BHyve   2020.0          Mumbai  Human Resources   
677  Saarthi Pedagogy   2015.0       Ahmadabad           EdTech   

                                          What_it_does  \
98   A real money game app specializing in trivia g...   
538                Soft Skills that make Smart Leaders   
551  A Future of Work Platform for diffusing Employ...   
677  India's fastest growing Pedagogy company, serv...   

                                              Founders  \
98                                              YC W21   
538  Holistic Development Programs for children in ...   
551                                  Backed by 100x.VC   
677                                           Pedagogy   

                            Investor Amount($)  \
98       Pritesh Kumar, Bharat Gupta  $1200000   
538                     Vishal Gupta   $300000   
551  Omkar Pandharkame, Ketaki Ogale   $300000   
677                   Sushil Agarwal  $1000000   

                               Stage  Funding_Year  
98                          Upsparks          2021  
538                     ah! Ventures          2021  
551   ITO Angel Network, LetsVenture          2021  
677  JITO Angel Network, LetsVenture          2021

In [38]:
# Change the values in the amount column with the original amount values in the Stage column
stage_list = ['Seed', 'Pre-series A', 'Series C']

stage_index = full_df.index[full_df['Amount($)'].isin(stage_list)]

full_df.loc[stage_index, ['Stage']] = full_df.loc[stage_index, ['Amount($)']].values

full_df.loc[stage_index, ['Amount($)']] = full_df.loc[stage_index, ['Investor']].values

full_df.loc[stage_index, ['Investor']] = full_df.loc[stage_index, ['Investor']].replace(stage_list, pd.NA, inplace=True)
full_df.loc[stage_index]

Company_Brand  Founded             HeadQuarter  \
242   Fullife Healthcare   2009.0  Pharmaceuticals\t#REF!   
257              MoEVing   2021.0         Gurugram\t#REF!   
545            AdmitKard   2016.0                   Noida   
1148           Godamwale   2016.0                  Mumbai   

                                                 Sector  \
242   Primary Business is Development and Manufactur...   
257   MoEVing is India's only Electric Mobility focu...   
545                                              EdTech   
1148                           Logistics & Supply Chain   

                                           What_it_does  \
242                                        Varun Khanna   
257                         Vikash Mishra, Mragank Jain   
545   A tech solution for end to end career advisory...   
1148  Godamwale is tech enabled integrated logistics...   

                                              Founders Investor  \
242                 Morgan Stanley Private Equity Asia     None   
257     Anshuman Maheshwary, Dr Srihari Raju Kalidindi     None   
545   Vamsi Krishna, Pulkit Jain, Gaurav Munjal\t#REF!     None   
1148         Basant Kumar, Vivek Tiwari, Ranbir Nandan     None   

           Amount($)         Stage  Funding_Year  
242        $22000000      Series C          2021  
257         $5000000          Seed          2021  
545         $1000000  Pre-series A          2021  
1148  1000000\t#REF!          Seed          2021

In [39]:
# Inspect the rows and columns
full_df.shape

(2600, 10)

In [40]:
# Checl null values in the columns
full_df.isnull().sum()

Company_Brand      0
Founded          700
HeadQuarter       97
Sector            16
What_it_does       0
Founders         539
Investor         616
Amount($)        299
Stage            783
Funding_Year       0
dtype: int64

A total of 299 firms didn't disclose or failed to provide information on the amount of funding received. We will proceed with dropping these values since there do not give us insight into the funding amount received.

In [41]:
# Drop missing values where the missing value is in the Amount column
full_df.dropna(subset='Amount($)', inplace=True)

In [42]:
full_df.shape

(2301, 10)

In [43]:
# Change the data type of the amount column to string and replace the $ and , with an empty string
full_df['Amount($)'] = full_df['Amount($)'].astype(str)

full_df['Amount($)'] = full_df['Amount($)'].apply(lambda x: x.replace('$',''))

full_df['Amount($)'] = full_df['Amount($)'].apply(lambda x: x.replace(',',''))

In [44]:
# Get the index position of the amount values that contain ₹
INR_index = full_df.index[full_df['Amount($)'].str.contains('₹')]
full_df.loc[INR_index, ['Amount($)']]

Amount($)
2354    ₹40000000
2355    ₹65000000
2359    ₹16000000
2360    ₹50000000
2368   ₹100000000
...           ...
2866  ₹1130000000
2867   ₹810000000
2869  ₹1400000000
2870    ₹10000000
2877    ₹35000000

[144 rows x 1 columns]

* According to [Exchange Rates UK](https://www.exchangerates.org.uk/INR-USD-spot-exchange-rates-history-2018.html), the average rate for 1 INR to USD in 2018 was `0.0146` and that is the rate we are using to convert the INR amount to dollars.

In [45]:
# Change the INR symbol and other special characters and replace them with an empty string
INR_index = full_df.index[full_df['Amount($)'].str.contains('₹')]

full_df['Amount($)'] = full_df['Amount($)'].apply(lambda x: x.replace('₹',''))
full_df['Amount($)'] = full_df['Amount($)'].apply(lambda x: x.replace('\t#REF!',''))

# Convert the amount column to a float datatype
full_df['Amount($)'] = full_df['Amount($)'].astype(float)

# Apply the exchange rate to the amount with the INR symbol
full_df.loc[INR_index, ['Amount($)']] = full_df.loc[INR_index, ['Amount($)']].values * 0.0146

full_df.loc[INR_index, ['Amount($)']]

Amount($)
2354    584000.0
2355    949000.0
2359    233600.0
2360    730000.0
2368   1460000.0
...          ...
2866  16498000.0
2867  11826000.0
2869  20440000.0
2870    146000.0
2877    511000.0

[144 rows x 1 columns]

### Check the headquarter column for inconsistencies

In [46]:
full_df['HeadQuarter'].unique()

array(['Bangalore', 'Mumbai', 'Gurugram', 'New Delhi', 'Hyderabad',
       'Jaipur', 'Ahmadabad', 'Chennai', None,
       'Small Towns, Andhra Pradesh', 'Goa', 'Rajsamand', 'Ranchi',
       'Faridabad, Haryana', 'Gujarat', 'Thane', 'Pune', 'Computer Games',
       'Cochin', 'Noida', 'Chandigarh', 'Gurgaon', 'Vadodara',
       'Food & Beverages', 'Pharmaceuticals\t#REF!', 'Gurugram\t#REF!',
       'Kolkata', 'Ahmedabad', 'Indore', 'Powai', 'Ghaziabad', 'Nagpur',
       'West Bengal', 'Patna', 'Samsitpur', 'Lucknow', 'Telangana',
       'Haryana', 'Silvassa', 'Faridabad', 'Ambernath', 'Panchkula',
       'Surat', 'Andheri', 'Telugana', 'Bhubaneswar', 'Kottayam',
       'Beijing', 'Panaji', 'Coimbatore', 'Satara', 'Orissia', 'Jodhpur',
       'New York', 'Santra', 'Trivandrum', 'Bhilwara', 'Kochi', 'London',
       'Information Technology & Services', 'The Nilgiris', 'Gandhinagar',
       'Belgaum', 'Tirunelveli, Tamilnadu', 'Singapore',
       'Jaipur, Rajastan', 'Delhi', 'California', '

* Foriegn values in the HeadQuarter column must be corrected
* Some HeadQuarters have values which are suppose to be trimmed down for consistency

In [47]:
sector_values = ['Information Technology & Services','Food & Beverages','Pharmaceuticals\t#REF!', 'Gurugram\t#REF!']
sector_values_index = full_df.index[full_df['HeadQuarter'].isin(sector_values)]
full_df.loc[sector_values_index]

Company_Brand  Founded                        HeadQuarter  \
241           MasterChow   2020.0                   Food & Beverages   
242   Fullife Healthcare   2009.0             Pharmaceuticals\t#REF!   
257              MoEVing   2021.0                    Gurugram\t#REF!   
1176                Peak   2014.0  Information Technology & Services   

                                                 Sector  \
241                                           Hauz Khas   
242   Primary Business is Development and Manufactur...   
257   MoEVing is India's only Electric Mobility focu...   
1176                     Manchester, Greater Manchester   

                                           What_it_does  \
241                 A ready-to-cook Asian cuisine brand   
242                                        Varun Khanna   
257                         Vikash Mishra, Mragank Jain   
1176  Peak helps the world's smartest companies put ...   

                                            Founders                Investor  \
241                    Vidur Kataria, Sidhanth Madan            WEH Ventures   
242               Morgan Stanley Private Equity Asia                    None   
257   Anshuman Maheshwary, Dr Srihari Raju Kalidindi                    None   
1176                                     Atul Sharma  SoftBank Vision Fund 2   

       Amount($)     Stage  Funding_Year  
241     461000.0      Seed          2021  
242   22000000.0  Series C          2021  
257    5000000.0      Seed          2021  
1176  75000000.0  Series C          2021

* At index position 242 and 257 the Investor, Founders, What_it_does and Sector values have been shifted to the left by one value and must be corrected
* At index position 1176, HeadQuarter value have been filled with Sector values which needs to be corrected
* From researching on the company names, it was realized that there were company's who had wrong values inputed and these values were corrected with the right information

In [48]:
full_df.loc[[32, 98, 241, 242, 1173, 1176, 257,1190,1393, 1483,2395, 2412]]

Company_Brand  Founded                        HeadQuarter  \
32                SuperK   2019.0        Small Towns, Andhra Pradesh   
98               FanPlay   2020.0                     Computer Games   
241           MasterChow   2020.0                   Food & Beverages   
242   Fullife Healthcare   2009.0             Pharmaceuticals\t#REF!   
1173            moneyHOP   2018.0                             London   
1176                Peak   2014.0  Information Technology & Services   
257              MoEVing   2021.0                    Gurugram\t#REF!   
1190             Prolgae   2016.0                       The Nilgiris   
1393             Dhurnia   2018.0                 Dhingsara, Haryana   
1483             Antaios   2016.0                             France   
2395              UClean      NaN                              India   
2412              Rooter      NaN                              India   

                                                 Sector  \
32                                               Retail   
98                                       Computer Games   
241                                           Hauz Khas   
242   Primary Business is Development and Manufactur...   
1173                                 Financial Services   
1176                     Manchester, Greater Manchester   
257   MoEVing is India's only Electric Mobility focu...   
1190                                      Biotechnology   
1393                                             EdTech   
1483                                       Tech company   
2395  Clean Energy, CleanTech, Laundry and Dry-cleaning   
2412                                             Sports   

                                           What_it_does  \
32    SuperK is a full-stack solution to empower sma...   
98    A real money game app specializing in trivia g...   
241                 A ready-to-cook Asian cuisine brand   
242                                        Varun Khanna   
1173   moneyHOP is India’s first cross-border neo bank.   
1176  Peak helps the world's smartest companies put ...   
257                         Vikash Mishra, Mragank Jain   
1190  Prolgae Spirulina Supplies Pvt. Ltd. is a Nord...   
1393  Developer of an online learning platform inten...   
1483  Developer of memory-based technology intended ...   
2395  UClean is a tech enabled laundry service provi...   
2412                  Rooter is a mobile-based platform   

                                            Founders  \
32                                      Neeraj Menta   
98                                            YC W21   
241                    Vidur Kataria, Sidhanth Madan   
242               Morgan Stanley Private Equity Asia   
1173                                    Mayank Goyal   
1176                                     Atul Sharma   
257   Anshuman Maheshwary, Dr Srihari Raju Kalidindi   
1190                                 Aakas Sadasivam   
1393                        Ajay Kumar, Murari Singh   
1483                            Jean Pierre Nozieres   
2395                                             NaN   
2412                                             NaN   

                                        Investor   Amount($)     Stage  \
32                                     STRIVE VC    800000.0      Seed   
98                   Pritesh Kumar, Bharat Gupta   1200000.0  Upsparks   
241                                 WEH Ventures    461000.0      Seed   
242                                         None  22000000.0  Series C   
1173                                        None   1200000.0      Seed   
1176                      SoftBank Vision Fund 2  75000000.0  Series C   
257                                         None   5000000.0      Seed   
1190                                     Vijayan    200000.0      Seed   
1393  Chandigarh Angels Network, Modulor Capital    100000.0      None   
1483                                        None  11000000.0      Non

* HeadQuarter values for these rows will be corrected with the right values

In [49]:
# Change the value of Sector to value of HeadQuarter value at index position 241
full_df.loc[241, ['Sector']] = full_df.loc[241, ['HeadQuarter']].values

# Change the values of Investor, Founders, and What_it_does to value with right values at index positions 242, and 257
full_df.loc[[242, 257], ['Investor']] = full_df.loc[[242, 257], ['Founders']].values
full_df.loc[[242, 257], ['Founders']] = full_df.loc[[242, 257], ['What_it_does']].values
full_df.loc[[242, 257], ['What_it_does']] = full_df.loc[[242, 257], ['Sector']].values

# Swap the Sector and HeadQuarter values at index position 1176
hq_value_1176 = full_df.loc[1176, ['Sector']].values
full_df.loc[[242, 1176], ['Sector']] = full_df.loc[[242, 1176], ['HeadQuarter']].values

full_df.loc[1176, ['HeadQuarter']] = hq_value_1176

# Correct wrong values in the column with the right values
correction_values = ['Bangalore', 'Bangalore', 'New Delhi', 'Mumbai', 'Bengaluru', 'Manchester', 'Bangalore', 'Nilgiris', 'Guruguram', 'Grenoble','Lucknow', 'New Delhi']
correction_values_index = [32, 98, 241, 242, 1173, 1176, 257,1190,1393, 1483,2395, 2412]
full_df.loc[correction_values_index, ['HeadQuarter']] = correction_values

# Clean special characters in the column
full_df['HeadQuarter'] = full_df['HeadQuarter'].str.replace('\\t#REF!', '', regex=True)

* Certain values in the HeadQuarter column have more information than need. After inspecting the values it was realized that the correct values are in the first postion of the string so this inconsistency must be corrected too.

In [50]:
# Get the values in the first position of the strings in all the rows
full_df['HeadQuarter'] = full_df['HeadQuarter'].str.split(',').str.get(0)

In [51]:
# View all rows with HeadQuarter value as Bangalore City
full_df.loc[full_df['HeadQuarter'] == 'Bangalore City']

Company_Brand  Founded     HeadQuarter  \
2476                           FeedMyPockets      NaN  Bangalore City   
2591                                   Trell      NaN  Bangalore City   
2688  BetterPlace Safety Solutions Pvt. Ltd.      NaN  Bangalore City   
2721                        Idha Skin Clinic      NaN  Bangalore City   
2842                                  GoGaga      NaN  Bangalore City   

                                                Sector  \
2476           Advertising, Human Resources, Marketing   
2591                                                 —   
2688               Human Resources, Security, Training   
2721  Beauty, Cosmetics, Health Care, Service Industry   
2842                 Dating, Private Social Networking   

                                           What_it_does Founders Investor  \
2476                        On Demand Staffing Platform      NaN      NaN   
2591  Trell is a location based network which helps ...      NaN      NaN   
2688  BetterPlace provides businesses with contracto...      NaN      NaN   
2721  Idha skin clinic jayanagar is dedicated to res...      NaN      NaN   
2842  GoGaga is a new age dating app that has remove...      NaN      NaN   

      Amount($)                  Stage  Funding_Year  
2476   642400.0                   Seed          2018  
2591  1250000.0                   Seed          2018  
2688  3000000.0               Series A          2018  
2721    73000.0                   Seed          2018  
2842    40000.0  Non-equity Assistance          2018

In [52]:
# Replace Bangalore City with Bangalore
bangalore_city_index =  full_df.index[full_df['HeadQuarter'] == 'Bangalore City']

full_df.loc[bangalore_city_index, ['HeadQuarter']] = 'Bangalore'

* Fill null values in the column with Unstated to help describe the data adequately

In [53]:
f"There are {full_df['HeadQuarter'].isnull().sum()} missing values in the HeadQuarter column and we will replace them with Unstated"

'There are 94 missing values in the HeadQuarter column and we will replace them with Unstated'

In [54]:
# Impute missing values with Unstated
imputer = SimpleImputer(missing_values=pd.NA, strategy='constant', fill_value='Unstated')
imputer.fit(full_df['HeadQuarter'].values.reshape(-1, 1))

SimpleImputer(fill_value='Unstated', missing_values=<NA>, strategy='constant')

In [55]:
# Replace the values in the position of the HeadQuarter column
full_df['HeadQuarter'] = imputer.transform(full_df['HeadQuarter'].values.reshape(-1, 1)).reshape(-1)

full_df['HeadQuarter'].isnull().sum()

0

### Check the Stage column for inconsistencies

Now, we have to correct the various stages. The assumption for this column are as follows:
* Pre-seed: This is the earliest stage of funding for a startup. Group all pre seed values under this value.
* Seed: This is the first round of institutional funding for a startup.
* Pre-series: This term is used to describe a funding round that is raised before a Series A round. Group all pre series values under this value
* Series A: This is the second round of institutional funding for a startup.
* Series B: This is the third round of institutional funding for a startup.
* Series C: This is the fourth round of institutional funding for a startup.
* Later Stage: Series D, Series E, Series F are subsequent rounds of institutional funding after a Series C round.
* Equity: This is a form of funding that is provided in exchange for an ownership stake in the company and it also constitutes PE and Grant.
* Others: Every other value that is present in the column.
* Unstated: These are undisclosed stages and null values in the column

In [56]:
# View unique variables in the Stage column
full_df['Stage'].unique()

array(['Pre-series A', None, 'Series D', 'Series C', 'Seed', 'Series B',
       'Series E', 'Pre-seed', 'Series A', 'Pre-series B', 'Debt',
       'Upsparks', 'Bridge', 'Seed+', 'Series F2', 'Series A+',
       'Series G', 'Series F', 'Series H', 'Series B3', 'PE', 'Series F1',
       'Pre-series A1', 'ah! Ventures', 'ITO Angel Network, LetsVenture',
       'Series D1', 'JITO Angel Network, LetsVenture', 'Seies A',
       'Pre-series', 'Series A2', 'Series I', 'Pre-series C', 'Series B2',
       'Pre- series A', 'Edge', 'Pre-Series B', 'Seed A', 'Series A-1',
       'Seed round', 'Seed Round & Series A', 'Pre Series A',
       'Pre series A1', 'Series E2', 'Pre series A', 'Seed Round',
       'Pre series B', 'Pre series C', 'Angel Round', 'Mid series',
       'Pre seed round', 'Seed funding', 'Seed Funding', nan,
       'Fresh funding', 'Post series A', 'Seed fund', 'Series B+',
       'Angel', 'Private Equity', 'Venture - Series Unknown', 'Grant',
       'Debt Financing', 'Post-IPO De

In [57]:
# Check null values in the column
f'''Considering we have {full_df['Stage'].isnull().sum()} missing values in the Stge column, replacing it with the mode will reduce the quality of our dataset, thus, we will replace the missing values with 'Unstated' to give wide perspective to the dataset.'''

"Considering we have 708 missing values in the Stge column, replacing it with the mode will reduce the quality of our dataset, thus, we will replace the missing values with 'Unstated' to give wide perspective to the dataset."

In [58]:
# Fill nulll values using simple imputer
imputer = SimpleImputer(missing_values=pd.NA, strategy='constant',  fill_value='Unstated')
imputer.fit(full_df['Stage'].values.reshape(-1, 1))

SimpleImputer(fill_value='Unstated', missing_values=<NA>, strategy='constant')

In [59]:
# Check for null values in the column
full_df['Stage'] = imputer.transform(full_df['Stage'].values.reshape(-1, 1)).reshape(-1)

full_df['Stage'].isnull().sum()

0

In [60]:
# Create a function which regroups the stage column values into our assumed values

def stage_cleaner(data, column_name='column'):
    stage_group = []
    for row in data[column_name]:
        if re.search(r'Pre[- ]?seed', row, re.IGNORECASE):
            stage_group.append('Pre seed')
        elif re.search(r'Seed|\bAngel\b', row, re.IGNORECASE):
            stage_group.append('Seed')
        elif re.search(r'Pre[- ]?ser(?:ies)?', row, re.IGNORECASE):
            stage_group.append('Pre series')
        elif re.search(r'Series[- ]?A(?:\D|$)', row, re.IGNORECASE):
            stage_group.append('Series A')
        elif re.search(r'Series[- ]?B(?:\D|$)', row, re.IGNORECASE):
            stage_group.append('Series B')
        elif re.search(r'Series[- ]?C(?:\D|$)', row, re.IGNORECASE):
            stage_group.append('Series C')
        elif re.search(r'Series[- ]?[DEFGH]', row, re.IGNORECASE):
            stage_group.append('Later stage')
        elif re.search(r'\bGrant\b|\bEquity\b|PE', row, re.IGNORECASE):
            stage_group.append('Equity')
        elif re.search(r'\bund(?:isclosed)?\b|\bUnstated\b', row, re.IGNORECASE):
            stage_group.append('Unstated')
        else:
            stage_group.append('Others')
    return stage_group

full_df['Stage'] = stage_cleaner(full_df, column_name='Stage')

In [61]:
# Check null values to ensure values have been regrouped
full_df['Stage'].unique()

array(['Pre series', 'Unstated', 'Later stage', 'Series C', 'Seed',
       'Series B', 'Pre seed', 'Series A', 'Others', 'Equity'],
      dtype=object)

### Check the Company column for inconsistencies

We will check for company's with duplicate values in the Investor, Amount, and Funding year columns in order to have duplicate inputs corrected.

In [62]:
full_df[full_df.duplicated(subset=['Company_Brand', 'Investor', 'Amount($)', 'Funding_Year'])]

Company_Brand  Founded HeadQuarter                    Sector  \
108     TartanSense   2015.0   Bangalore    Information Technology   
110       Kirana247   2018.0   New Delhi  Logistics & Supply Chain   
113   IsGoingOnline   2015.0      Cochin                    Retail   
114        Camp K12   2010.0    Gurugram                    EdTech   
115     Elda Health   2020.0   Bangalore                Healthcare   
116          Sporjo   2020.0      Mumbai                SportsTech   
118     SuperGaming   2017.0        Pune                    Gaming   
277         Bizongo   2015.0      Mumbai             B2B Ecommerce   
416           Siply   2020.0   Bangalore                   FinTech   
1076   Ola Electric   2017.0   Bangalore                Automotive   
1318     Credgenics   2018.0    Gurugram                   FinTech   
1994    Rebel Foods      NaN    Unstated                 Food tech   
2173           Yolo   2019.0    Gurugram                 Transport   

                                           What_it_does  \
108   TartanSense unlocks value for small farm holde...   
110   An on-demand FMCG supply chain company leverag...   
113   Kochi-based digital catalogue and marketplace ...   
114   Camp K12 is a global online school for 21st ce...   
115   Elda Health is a digital wellness platform tha...   
116   A one stop destination to build a successful c...   
118   At SuperGaming, building India's best gaming c...   
277   izongo is India’s largest B2B platform for mad...   
416   Siply Services operates as a micro-savings pla...   
1076  Charging ahead to accelerate the world’s trans...   
1318  Credgenics is a tech-enabled platform backed b...   
1994                           It runs internet kitchen   
2173                   Intercity transportation startup   

                                              Founders  \
108                                       Jaisimha Rao   
110                Tarun Jiwarajka, Pankhuri Jiwarajka   
113                                         Eobin Alex   
114                                       Anshul Bhagi   
115   Swathi Kulkarni, Sindhuri Ananth, Shubham Sharma   
116                                      G Srinivvasan   
118         Roby John, Sanket Nadhani, Navneet Waraich   
277            Sachin Agrawal, Aniket Deb, Ankit Tomar   
416                               Sousthav Chakrabarty   
1076                                  Bhavish Aggarwal   
1318         Anand Agrawal, Mayank Khera, Rishabh Goel   
1994                  Jaydeep Barman, Kallol Bannerjee   
2173         Shailesh Gupta, Mukul Shah, Danish Chopra   

                                               Investor    Amount($)  \
108                       FMC, Omnivore, Blume Ventures    5000000.0   
110                                                None    1000000.0   
113      Unicorn India Ventures, SEA Fund, Devdatt Shah     200000.0   
114            Matrix Partners India, Elevation Capital   12000000.0   
115                      Avaana Capital, Orios Ventures    1500000.0   
116                                         Punit Balan    2000000.0   
118   Skycatcher, AET Fund, BAce Capital, Dream Incu...    5500000.0   
277                                        Tiger Global  100000000.0   
416                           Inflection Point Ventures    1000000.0   
1076                              Falcon Edge, Softbank  200000000.0   
1318                    Accel India, Breadlake Ventures    3000000.0   
1994                                  Coatue Management   50000000.0   
2173                                        Undisclosed     600000.0   

            Stage  Funding_Year  
108      Series A          2021  
110    Pre series          2021  
113    Pre series          2021  
114      Series A          2021  
115          Seed          2021  
116    Pre series          2021  
118      Series A          2021  
277   Later stage          2021  
416          Seed          2021  
1076     Unstated          2021  
13

In [63]:
# Remove the duplicate values found
full_df.drop_duplicates(subset=['Company_Brand', 'Investor', 'Amount($)', 'Funding_Year'], keep='first', inplace=True)

### Check the Sector column for inconsistencies

In [64]:
full_df['Sector'].unique()

array(['AI startup', 'EdTech', 'B2B E-commerce', 'FinTech',
       'Home services', 'HealthTech', 'E-commerce', 'B2B service',
       'Helathcare', 'Renewable Energy', 'Electronics', 'IT startup',
       'Food & Beverages', 'Aeorspace', 'Deep Tech', 'Dating', 'Gaming',
       'Robotics', 'Retail', 'Food', 'Oil and Energy', 'Tech Startup',
       'AgriTech', 'Telecommuncation', 'Milk startup', 'AI Chatbot', 'IT',
       'Logistics', 'Hospitality', 'Fashion', 'Marketing',
       'Transportation', 'LegalTech', 'Food delivery', 'Automotive',
       'SaaS startup', 'Fantasy sports', 'Video communication',
       'Social Media', 'Skill development', 'Rental', 'Recruitment',
       'Sports', 'Computer Games', 'Consumer Goods',
       'Information Technology', 'Apparel & Fashion',
       'Logistics & Supply Chain', 'Healthcare', 'SportsTech', 'HRTech',
       'Wine & Spirits', 'Mechanical & Industrial Engineering',
       'Spiritual', 'Financial Services', 'Industrial Automation',
       'Heal

Convert the values into capitalize string format to ensure consistency in the values

In [65]:
full_df['Sector'] = full_df['Sector'].str.capitalize()

In [66]:
full_df['Sector'].value_counts()

Sector
Fintech                                       219
Edtech                                        171
E-commerce                                     69
Financial services                             69
Agritech                                       55
                                             ... 
Digital tech                                    1
Eye wear                                        1
Iot/automobile                                  1
Nutrition tech                                  1
Biotechnology, health care, pharmaceutical      1
Name: count, Length: 686, dtype: int64

Select the first index position of values in the column which have multiple values separated by a comma

In [67]:
full_df['Sector'] = full_df['Sector'].str.split(',').str.get(0)

In [68]:
full_df['Sector'].unique()

array(['Ai startup', 'Edtech', 'B2b e-commerce', 'Fintech',
       'Home services', 'Healthtech', 'E-commerce', 'B2b service',
       'Helathcare', 'Renewable energy', 'Electronics', 'It startup',
       'Food & beverages', 'Aeorspace', 'Deep tech', 'Dating', 'Gaming',
       'Robotics', 'Retail', 'Food', 'Oil and energy', 'Tech startup',
       'Agritech', 'Telecommuncation', 'Milk startup', 'Ai chatbot', 'It',
       'Logistics', 'Hospitality', 'Fashion', 'Marketing',
       'Transportation', 'Legaltech', 'Food delivery', 'Automotive',
       'Saas startup', 'Fantasy sports', 'Video communication',
       'Social media', 'Skill development', 'Rental', 'Recruitment',
       'Sports', 'Computer games', 'Consumer goods',
       'Information technology', 'Apparel & fashion',
       'Logistics & supply chain', 'Healthcare', 'Sportstech', 'Hrtech',
       'Wine & spirits', 'Mechanical & industrial engineering',
       'Spiritual', 'Financial services', 'Industrial automation',
       'Heal

In [79]:
# Check null values in the column
f'''Considering we have {full_df['Sector'].isnull().sum()} missing values in the Stge column, replacing it with the mode will reduce the quality of our dataset, thus, we will replace the missing values with 'Unstated' to give wide perspective to the dataset.'''

"Considering we have 16 missing values in the Stge column, replacing it with the mode will reduce the quality of our dataset, thus, we will replace the missing values with 'Unstated' to give wide perspective to the dataset."

In [80]:
# Impute nulll values using simple imputer
imputer = SimpleImputer(missing_values=pd.NA, strategy='constant',  fill_value='Unstated')
imputer.fit(full_df['Sector'].values.reshape(-1, 1))

SimpleImputer(fill_value='Unstated', missing_values=<NA>, strategy='constant')

In [81]:
# Fill null values in the column
full_df['Sector'] = imputer.transform(full_df['Sector'].values.reshape(-1, 1)).reshape(-1)

full_df['Sector'].isnull().sum()

0

In [201]:
full_df.drop(['Assigned Sector', 'New'], axis=1, inplace=True)

In [238]:
a =full_df[full_df['sect'] == 'Unstated']
a['Sector'].unique()

array(['Construction', 'Education management', 'Music', 'Cosmetics',
       'Mobility', 'Content publishing', 'Celebrity engagement',
       'Cannabis startup', 'Blogging', 'Home decor', 'Taas startup',
       'Vehicle repair startup', 'Pollution control equiptment',
       'Consulting', 'Fitness', 'D2c', 'D2c business', 'Human resources',
       'Online storytelling', 'Crm', 'Cultural', 'Clothing',
       'Business supplies & equipment', 'Nft',
       'Professional training & coaching', 'Maritime', 'Higher education',
       'Translation & localization', 'Foootwear', 'Veterinary',
       'Education', 'Management consulting', 'Product studio',
       'Staffing & recruiting', 'Escrow', 'Crowdsourcing',
       'Work fulfillment', 'Job portal', 'Paas startup', 'Qsr startup',
       'Fm', 'Video', 'Fitness startup',
       'Multinational conglomerate company', 'Eyeglasses', 'Car service',
       'Service industry', 'Publication', 'Car trade',
       'Wl & rac protection', 'Home interior se

Group the sector column into the 11 major industry sectors based on GSIC standards.

In [236]:
# Create a function which assigns industry sector to the values in the columnn

def assign_sector_to_values_regex(dataframe, column_name='column'):
    assigned_sectors = []
    for row in dataframe[column_name]:
        if re.search(r'\b(AI startup|.*Tech.*|.*tech|Robotics|IT startup|Deep.*|.*ware.*|SaaS.*|A.*|Biot.*|Nanotechnology|.*ata.*|Tech Startup|Deep.*|Drone|Computer software|Ott|Mobile games|Information services|Enterprise resource planning \(erp\)|Technology|Networking|Software|Computer software|Digital platform|Software company|.*esign|Digital mortgage|E-l.*|E-m.*|Hr.*|Analy.*|Recruitment|Skill development|Internet|.*platform|.*intelligence|Apps|Information Technology & Services|Iot|Software Startup|Computer & Network Security|Software|Innovation management|It|Cloud.*)\b', row, re.IGNORECASE):
            assigned_sectors.append('Information Technology')
        elif re.search(r'\b(Hel.*|Hea.*|Hosp.*|.*care|Pharma.*|Biopharma|Medical)\b', row, re.IGNORECASE):
            assigned_sectors.append('Health Care')
        elif re.search(r'\b(Fin.*|Banking|Capital Markets|Insurance|Insuretech|Cryptocurrency|Marketing|Equity Management|Investment.*|Insurance tech|Virtual banking|Wealth management|Credit.*|Crypto|Legal.*|.*chain|.*capital.*|Mutual funds|Consultancy|Taxation|Cybersecurity)\b', row, re.IGNORECASE):
            assigned_sectors.append('Financials')
        elif re.search(r'\b(B2.*|.*commerce.*|Home services|Retail|Dating|Gaming|Fashion.*|.*well.*|Spiritual|Matrimony|Beauty products|Tourism|Wholesale|Femtech|Co-working|Fantasy.*|Basketbal|Retail tech|Sport.*|Consumer appliances|Beauty|Consumer|Nut.*)\b', row, re.IGNORECASE):
            assigned_sectors.append('Consumer Discretionary')
        elif re.search(r'\b(Aer.*|Aeo.*|Renew.*|Electronics|AgriTech|Transportation|Automotive|Manufacturing startup|Manufacturing|Automobile|Automobiles|Logist.*|Mechanical & Industrial Engineering|Industrial Automation|CleanTech|Automotives|Renewable player|Technology|Automation|Automation tech|Renewable|Construction tech|Defense tech|Transport automation|Industrial|Environmental consulting|Commercial)\b', row, re.IGNORECASE):
            assigned_sectors.append('Industrials')
        elif re.search(r'\b(.*commun.*|.*media.*|Broadcasting|Advertising|Content management|Audio|Search engine|.*entertainment|Social.*|Events|Digital platform|Online media|Video sharing platform|Market research|Enterprise resource planning \(erp\)|Classifieds|Crowdfunding|Computer)\b', row, re.IGNORECASE):
            assigned_sectors.append('Communication Services')
        elif re.search(r'\b(.*nergy.*|EV|Solar.*|Ev startup|Electricity|Clean energy|Energy)\b', row, re.IGNORECASE):
            assigned_sectors.append('Energy')
        elif re.search(r'\b(.*lifestyle|Textiles|Eyewear|Merchandise|Biotech|Biomaterial startup|Battery design|Battery)\b', row, re.IGNORECASE):
            assigned_sectors.append('Materials')
        elif re.search(r'\b(Real Estate|Housing Marketplace|Furniture Rental|Real estate|Housing|Marketplace|Rental|Furniture|Rental space|Proptech|Commercial real estate)\b', row, re.IGNORECASE):
            assigned_sectors.append('Real Estate')
        elif re.search(r'\b(Water purification|Facilities Services|Environmental Services|Hygiene|Sanitation solutions|CleanTech|Battery manufacturer)\b', row, re.IGNORECASE):
            assigned_sectors.append('Utilities')
        elif re.search(r'\b(Food.*|Craft beer|.*verage.*|Appliance|Basketbal|Cooking|Milk startup|Farming|Fishery|Dental|Catering|Consumer lending|Wine.*|Fmcg|Customer service|Tobacco|Delivery|Dietary supplements|Last mile transportation|Online portals|Retail startup|Customer service company|Content marketplace|E-mobility|E-store|E-market|Agritech startup|Api platform|Agriculture|E-connect|Dairy.*|Health and fitness|Consumer)\b', row, re.IGNORECASE):
            assigned_sectors.append('Customer Staples')
        else:
            assigned_sectors.append('Unstated')
    return assigned_sectors


full_df['sect'] = assign_sector_to_values_regex(full_df, column_name='Sector')
full_df.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      Ai startup   
1          upGrad   2015.0      Mumbai          Edtech   
2     Lead School   2012.0      Mumbai          Edtech   
3         Bizongo   2015.0      Mumbai  B2b e-commerce   
4        FypMoney   2021.0    Gurugram         Fintech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor    Amount($)        Stage  \
0                    BEENEXT, Entrepreneur First    1200000.0   Pre series   
1       Unilazer Ventures, IIFL Asset Management  120000000.0     Unstated   
2               GSV Ventures, Westbridge Capital   30000000.0  Later stage   
3                         CDC Group, IDG Capital   51000000.0     Series C   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    2000000.0         Seed   

   Funding_Year                    sect  
0          2021  Information Technology  
1          2021  Information Technology  
2          2021  Information Technology  
3          2021  Consumer Discretionary  
4          2021  Information Technology